build.gradle

```diff
@@ -103,6 +107,30 @@ javadoc {
     }
 }
 
+
+if (rootProject.hasProperty("releaseMode")) {
+    gitPublish {
+
+        repoUri = 'https://github.com/rpmcdougall/RxJava.git'
+
+        // branch will be created if it doesn't exist
+        branch = 'gh-pages-test'
+
+        contents {
+            from 'build/docs'
+        }
+
+        preserve {
+            include '1.x/**'
+            include '2.x/**'
+            include 'images/**'
```

Comment:
There is an `rx` directory here to be preserved as well: https://github.com/ReactiveX/RxJava/tree/gh-pages/javadoc